### PR TITLE
Vector3.unproject(): Avoid computing inverse

### PR DESCRIPTION
**Pull request by WestLangley**
As suggested in https://github.com/mrdoob/three.js/pull/14685#issuecomment-414067093.

...plus it removes another annoying cyclic reference in the build. :-)

In some examples, `Vector3.unproject()` is recomputing  the matrix inverse every frame.

I think we are safe making this change now, but we need to be careful when using `projectionMatrix.copy()`.

@Mugen87 What do you think?

### CODE REVIEW

File: src/math/Vector3.js [hunk 1]
```diff
@@ -320,15 +319,9 @@ Object.assign( Vector3.prototype, {
 
 	unproject: function () {
 
-		var matrix = new Matrix4();
+		return this.applyMatrix4( camera.projectionMatrixInverse ).applyMatrix4( camera.matrixWorld );
```
Comment (Mugen87) [inline src/math/Vector3.js]: It looks like `camera` is missing as a parameter in the method signature.

**Comment by Mugen87** [comment]
I think it's okay to do this change since we also assume in `Vector3.project()` that `matrixWorldInverse` is with the world matrix of the camera.

File: src/math/Vector3.js (hunk 1 above)
Comment (WestLangley) [reply src/math/Vector3.js]: Ugh... it worked in testing because `camera` is global... Thanks for noticing...

Comment (WestLangley) [reply src/math/Vector3.js]: Fixed.

**Comment by WestLangley** [comment]
@Mugen87 Search for `projectionMatrix.copy` and notice where it is used without copying the inverse, too. This is what I was referring to...

**Comment by Mugen87** [comment]
I see what you mean. In the examples, the copy happens in `Reflector`, `Refractor`, `Water` and `CubeTexturePass`. And in `StereoCamera` and `CameraHelper` which are located in the core. But non of these classes uses `Vector3.unproject()`.

**Comment by WestLangley** [comment]
@Mugen87 I think we need this subtle change in the comment and code:

```js
// we need just camera projection matrix inverse
// world matrix must be identity

camera.projectionMatrixInverse.copy( this.camera.projectionMatrixInverse );
```

**Review by Mugen87 — APPROVED** [review]

**Comment by mrdoob** [comment]
Thanks!

**Comment by bleighb** [comment]
I don't really understand what this PR is about to be honest, but I've isolated this PR as the one that breaks raycasting in my app. 

I am using `projectionMatrix.copy` because I'm using a hybrid camera that transitions between the two. Any advice for how to fix this? Here's probably the most relevant part of my code

```js
const matrixToCopy = this.isUsingOrthographicCamera ?
    this._orthographicCamera.projectionMatrix :
    this._perspectiveCamera.projectionMatrix;

this._hybridCamera.projectionMatrix.copy(matrixToCopy);
```

And the full file is here: https://gist.github.com/bleighb/83ff6814a7c8968425d48db0f580f442

**Comment by bleighb** [comment]
Pinging @Mugen87 @WestLangley Any help would be greatly appreciated 😊

**Comment by WestLangley** [comment]
@bleighb This is not a help site. Please use the three.js [forum](https://discourse.threejs.org) instead.

**Comment by bleighb** [comment]
Sorry about that. I'm not asking for handouts, it's just I took a good amount of time to isolate the exact commit that broke my app and I figured other people might be affected by this bug. Since you made the change, I thought the answer might either indicate that either it was a mistake on your part or I'm doing something terribly wrong 😄

**Comment by bleighb** [comment]
I just fixed it by copying over the `projectionMatrixInverse` when I switch between camera types. 

You mentioned before:
>Thing is, projectionMatrixInverse is computed in updateProjectionMatrix() only; it is not set elsewhere. For example, it is not set whenever projectionMatrix.copy() is used. This will cause the matrix and its inverse to get out-of-sync.

>We need to come up with an approach that will ensure they are synchronized if we want to make use of projectionMatrixInverse.

I believe this is what happened to me. The matrices came out of sync and it yielded weird side effects. I'm not asking for help anymore - I'm just trying to see if this is an issue or could be improved.

**Comment by WestLangley** [comment]
Good. You were able to solve the bug in your app yourself. I think the library is fine as is.

**Pull request merged**